src/components/task-list/task-list.component.js

```diff
@@ -28,10 +26,11 @@ const TaskList = (props) => {
             <h4>Priority:</h4>
             <h5>{task.priority}</h5>
             <input
+              checked={task.completed ? true : false}
               hover-message={task.hoverMessage}
               className={styles.checkbox}
-              onMouseEnter={hoverCheckbox}
               type="checkbox"
+              onChange={toNotHaveError}
```

Comment:
make this component take a prop `handleTaskCompleted` (or similar) that takes an `id`.

The definition of `handleTaskCompleted` (in your `task-app.component`) would have _something like a find_ based on said `id`, and when a `task` is found you'd proceed to update the `completed` field.

Note that I've referred to an `id` that is not currently present in the project. I would suggest adding some sort of unique identifier [when each task is created](https://github.com/Joao-Matias/to-do-list/blob/TD-02/add-new-task-functionality/src/components/task-form/task-form.component.js#L23) for the competition of this PR.

it can be as simple as a:

```typescript
    addNewTask({ ...task, completed: false, id: Date.now() });
```

or after installing [`uuidv4`](https://www.npmjs.com/package/uuidv4), it can be like:
```typescript
// top of the file
import { uuid } from 'uuidv4';

    // content of the `submitForm` method:
    addNewTask({ ...task, completed: false, id: uuid() });
```

Reply:
Also, it seems like your [second PR](https://github.com/Joao-Matias/to-do-list/blob/TD-02/add-new-task-functionality/src/components/task-form/task-form.component.js#L23) hasn't been merged to [main](https://github.com/Joao-Matias/to-do-list/blob/main/src/components/task-form/task-form.component.js#L23)... (I can't see the changes in the `addNewTask` function)

Reply:
9e321c69f491559e9b9770e6c97051e39a21e54d Added id to each task and attached callback to checkbox compare id with target and change state based on it.